### PR TITLE
[50727] Android: Changed logic so user is navigated to main page if denying key upload

### PR DESCRIPTION
**Pull request by jonasofrich**

### CODE REVIEW

**Review by pavshr — APPROVED** [review]
Approved :)

File: NDB.Covid19/NDB.Covid19.Droid/Views/AuthenticationFlow/LoadingPageActivity.cs [hunk 1]
```diff
@@ -74,16 +69,21 @@ void OnActivityFinished()
...
         {
-            AuthErrorUtils.GoToTechnicalError(this,LogSeverity.ERROR, e, "Pushing keys failed" );
+            if (e is AccessDeniedException)
+            {
+                LogUtils.LogMessage(LogSeverity.INFO, "The user refused to share keys", null);
+                GoToInfectionStatusPage();
+            }
+            else
+            {
+                RunOnUiThread(() => AuthErrorUtils.GoToTechnicalError(this, LogSeverity.ERROR, e, "Pushing keys failed"));
```
Comment (nc-kano) [inline NDB.Covid19/NDB.Covid19.Droid/Views/AuthenticationFlow/LoadingPageActivity.cs:81]: probably goToInfectionStatusPage should also be done on UI thread

Comment (Frederik-S-N-dk) [reply NDB.Covid19/NDB.Covid19.Droid/Views/AuthenticationFlow/LoadingPageActivity.cs:81]: @nc-kano Should Jonas do anything in this regard, or will we just merge this PR into dev?

Comment (nc-kano) [reply NDB.Covid19/NDB.Covid19.Droid/Views/AuthenticationFlow/LoadingPageActivity.cs:81]: yes, just needs to add RunOnUIThread :)

Comment (Frederik-S-N-dk) [reply NDB.Covid19/NDB.Covid19.Droid/Views/AuthenticationFlow/LoadingPageActivity.cs:81]: @jonasofrich see above :)

Comment (jonasofrich) [reply NDB.Covid19/NDB.Covid19.Droid/Views/AuthenticationFlow/LoadingPageActivity.cs:81]: Done

**Review by nc-kano — APPROVED** [review]

**Pull request merged**